### PR TITLE
Vampire hunter clothes for chaplains + whip changes

### PR DESCRIPTION
**Pull request by ghost**
[!!VIDEO DEMONSTRATING WHIP CHANGES!!](http://fluffyboys.moe/v0nTIR3QIi.mp4)

dreamseeker_Qbq7tkfjXY

Changes whips to use hookshot behavior instead of beam projectiles. They can't grab things, this change is (mostly) cosmetic.

Also adds a new set of clothing for chaplains, and a bonus toy to the PietyVend's premium selection. The whip changes were needed solely to add this toy. Icons by Edwarius/Sage of Light.

:cl:
 * rscadd: New vampire hunter-themed chaplain gear.
 * tweak: Whips are now more nifty.

### CODE REVIEW

**Comment by Irradiation** [comment]
Hurray for not blowing up a weldertank because you whipped it good.

**Comment by SageOfLight** [comment]
What a horrible night to have a curse, bruh. 

I made the sprites. If people enjoy this I may create some other Castlevania themed paraphernalia.

File: code/modules/projectiles/projectile/hookshot.dm [hunk 1]
```diff
@@ -174,3 +180,42 @@
...
+	if(W.firer?.is_wearing_item(/obj/item/clothing/head/energy_dome) && whipitgood_bonus)
+		force += whipitgood_bonus
+		visible_message("<span class='warning'>[W.firer] whips it good!</span>")
+	..(atarget, blocked)
+
+/obj/item/projectile/hookshot/whip/vampkiller
+	name = "flail"
+	icon_state = "vampkiller"
+	icon_name = "vampkiller"
+	damage = 1
```
Comment (boy2mantwicethefam) [inline code/modules/projectiles/projectile/hookshot.dm]: Doesn't this make the flail deal 1 damage per hit?

**Comment by boy2mantwicethefam** [comment]
Got a gif or video on the new whip & flails?
Edit: I have not seen it. I have failed and this is my shame
Obligatory https://www.youtube.com/watch?v=JYfgpodCVHE

File: code/modules/projectiles/projectile/hookshot.dm (hunk 1 above)
Comment (ghost) [reply code/modules/projectiles/projectile/hookshot.dm]: Yes, that's intentional since this version of the item is easily obtainable from the pietyvend

Comment (boy2mantwicethefam) [reply code/modules/projectiles/projectile/hookshot.dm]: Shrug

**Comment by SirSkeleto** [comment]
the holy one just does 30 brute
30 is a lot, especially for free at a range on a fairly quick cooldown
maybe <30 for humans and >30 for vamps?
just a suggestion

**Comment by ghost** [comment]
I'm gonna add in a holy check for projectiles tomorrow so consider that a placeholder for now

**Comment by BomberBro** [comment]
I like it, but man those are some big item sprites, they need to be reduced a bit.

**Comment by SonixApache** [comment]
if those are the item icons then they are way too fucking big

**Comment by SageOfLight** [comment]
> if those are the item icons then they are way too fucking big

image
_Fine._

**Comment by ghost** [comment]
ok the icons are smallified, @SonixApache remove your confused emoji or I'm going to end you

Also lowered the damage for the regular whip since the hookshot can be spammed relatively fast in close quarters. The real vampire killer also does less damage unless it's attacking a vampire now.

**Comment by SonixApache** [comment]
you got my :+1:

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -3014,6 +3016,7 @@ var/global/num_vending_terminals = 1
 		/obj/item/weapon/reagent_containers/food/drinks/bottle/holywater = 1,
 		/obj/item/clothing/head/helmet/knight/templar = 2,
  		/obj/item/clothing/suit/armor/knight/templar = 5,
+		/obj/item/weapon/gun/hookshot/whip/vampkiller = 1,
```
Comment (jknpj) [inline code/game/machinery/vending.dm:3019]: are you sure you want to add a weapon to the pietyvend?

Comment (ghost) [reply code/game/machinery/vending.dm:3019]: It does 1 damage, if that's a problem I can make it do 0.

**Comment by ghost** [comment]
Threw in a vampire hunter religion with bible icon and fluff once again by Edwarius/Sage of Light. Starts you out with the vampire hunter gear sans whip.

I also reduced the damage of the "toy" version's projectile to 0 to placate the c*llabs. The item itself has been changed to have 15 force, same as the nullrod.

**Comment by GermanSanta** [comment]
Too op. Needs to be balanced by the removal of pants.
external-content duckduckgo

**Comment by SageOfLight** [comment]
Fake news. Also, you're probably a vampire who wants to bite my beefy thighs.

image

You can be pantsless by wearing a proper jumpsuit, or lack of one.

image

**Pull request merged**